core/lib/via_btc_client/src/inscriber/mod.rs

```diff
@@ -1 +1,734 @@
+            value: tx_input_data.unlocked_value - fee_amount,
+            script_pubkey: self.signer.get_p2wpkh_script_pubkey().clone(),
+        };
+
+        let res = CommitTxOutputRes {
+            commit_tx_change_output,
+            commit_tx_tapscript_output: inscription_commitment_output,
+            commit_tx_fee_rate: fee_rate,
+            _commit_tx_fee: fee_amount,
+        };
+
+        Ok(res)
+    }
+
+    async fn get_fee_rate(&self) -> Result<u64> {
+        let res = self.client.get_fee_rate(FEE_RATE_CONF_TARGET).await?;
+
+        Ok(res)
+    }
+
+    fn sign_commit_tx(
+        &self,
+        input: &CommitTxInputRes,
+        output: &CommitTxOutputRes,
+    ) -> Result<FinalTx> {
+        let temp_tx_out1 = TxOut {
+            value: Amount::ZERO,
+            script_pubkey: ScriptBuf::default(),
+        };
+        let temp_tx_out2 = TxOut {
+            value: Amount::ZERO,
+            script_pubkey: ScriptBuf::default(),
+        };
+
+        let mut commit_outputs: [TxOut; 2] = [temp_tx_out1, temp_tx_out2];
+
+        commit_outputs[COMMIT_TX_CHANGE_OUTPUT_INDEX as usize] =
+            output.commit_tx_change_output.clone();
+        commit_outputs[COMMIT_TX_TAPSCRIPT_OUTPUT_INDEX as usize] =
+            output.commit_tx_tapscript_output.clone();
+
+        let mut unsigned_commit_tx = Transaction {
+            version: transaction::Version::TWO,  // Post BIP-68.
+            lock_time: absolute::LockTime::ZERO, // Ignore the locktime.
+            input: input.commit_tx_inputs.clone(),
+            output: commit_outputs.to_vec(), // Outputs, order does not matter.
+        };
+
+        let sighash_type = EcdsaSighashType::All;
+        let mut commit_tx_sighasher = SighashCache::new(&mut unsigned_commit_tx);
+
+        let script_pubkey = self.signer.get_p2wpkh_script_pubkey();
+        for (index, _input) in input.commit_tx_inputs.iter().enumerate() {
+            let sighash = commit_tx_sighasher
+                .p2wpkh_signature_hash(
+                    index,
+                    script_pubkey,
+                    input.utxo_amounts[index],
+                    sighash_type,
+                )
+                .context("Failed to create sighash")?;
+
+            // Sign the sighash using the signer
+            let msg = Message::from(sighash);
+            let signature = self.signer.sign_ecdsa(msg)?;
+
+            // Update the witness stack.
+            let signature = bitcoin::ecdsa::Signature {
+                signature,
+                sighash_type,
+            };
+            let pk = self.signer.get_public_key();
+
+            *commit_tx_sighasher
+                .witness_mut(index)
+                .ok_or_else(|| anyhow::anyhow!("Failed to get witness"))? =
+                Witness::p2wpkh(&signature, &pk);
+        }
+
+        let commit_tx = commit_tx_sighasher.into_transaction();
+
+        let res = FinalTx {
+            tx: commit_tx.clone(),
```

Comment:
let's change it like
```
let txid = commit_tx.compute_txid();
```
and don't clone the entire tx

Reply:
`let commit_tx = commit_tx_sighasher.into_transaction();`
        
this method return an `mut refrence `to real transaction and i need real tx for my struct so i should clone it.
also i could deference it but i think the cloning is more readable.